### PR TITLE
Fix opencv on Apple M1

### PR DESCRIPTION
**Pull request by sunnycase**
Specify library name and version:  **opencv/4.5.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/opencv/4.x/conanfile.py [hunk 1]
```diff
@@ -321,7 +321,7 @@ def add_components(components):
                             self.cpp_info.components[conan_component].libdirs.append("lib")
                             self.cpp_info.components[conan_component].libs += tools.collect_libs(self)
 
-                if self.settings.os == "iOS":
+                if self.settings.os == "iOS" or tools.os_info.is_macos:
```
Comment (jgsogo) [inline recipes/opencv/4.x/conanfile.py:324]: Hi, `tools.os_info` returns information about the _build_ machine, while `package_info` should return information about the _host_ libraries that the consumer is going to use/link. In a cross-building scenario we might be running on MacOS but compiling binaries for other OS, in that scenario we don't want this condition to evaluate to `True`. 

Hope this distinction is clear.

Comment (sunnycase) [reply recipes/opencv/4.x/conanfile.py]: Fixed.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by jgsogo** [comment]
Conflicts! 👮‍♂️

**Comment by prince-chrismc** [comment]
@sunnycase please rebase the PR 🙏

**Comment by sunnycase** [comment]
@prince-chrismc Rebased.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**